### DISCUSSION TITLE
Unable to Bind timebase stream to java class in latest version 6.1.14

### DISCUSSION THREAD

**Discussion by ShivaniSK** [finos/TimeBase-CE#244] [Q&A] [answered]
Hi @alex-karpovich ,
We recently upgraded timebase server from 6.1.12. to 6.1.14 version.
The streams created in timebase 6.1.14 are same as 6.1.12 timebase instance. However, we are unable to bind the java class to the stream in the latest version. 

getting below error

```
2023-07-27 13:16:31.247 WARN [main] Bind error: type loader is unable to load class. (will try mapping parent): 
java.lang.ClassNotFoundException: QUERY78375362613
```

Stream with name "test" is created in timebase.
The "test" stream has a column "venue" with VARCHAR datatype and corresponding TestMessage model class is created with "venue" field having String datatype.

```
        DXTickDB db = TickDBFactory.createFromUrl ("dxtick://<url>");
        db.open (false);
        long endTime =1661385612410L;

        SelectionOptions options = new SelectionOptions();

        options.typeLoader = new SimpleTypeLoader(null, TestMessage.class);
        options.raw = false;

        String queryF = "select venue from test";
      
        try (InstrumentMessageSource cursor = db.executeQuery(queryF, options)){
            while (cursor.next()) {
                TestMessage bboTest = (TestMessage) cursor.getMessage();
                System.out.println(bboTest);
                if (cursor.getMessage().getTimeStampMs() >= endTime) {
                    break;
                }
            }
        } finally {
            db.close();
        }

```

The above mapping works fine with old timebase serevr version 6.1.12. But fails in 6.1.14.

Can you please help us out. Are there any "typeLoader" related changes in the latest version?

We are using Java 11 and gradle version 7.3.3

Regards,
Shivani

**Comment by alex-karpovich** [comment, marked as answer]
Greetings @ShivaniSK 

Due to massive changes in QQL, now queries has names like "QUERY<radom_name>"
so you need to use next
```
options.typeLoader = new SimpleTypeLoader() {
            @Override
            public Class<?> load(ClassDescriptor cd) throws ClassNotFoundException {
                if (cd.getName().startsWith("QUERY"))
                    return TestMessage.class;
                return super.load(cd);
            }
        };
```

and please use 6.1.17

**Reply by ShivaniSK** [reply]
Thanks @alex-karpovich this helps! Also if you can point us to the documentation or release notes it would be of great help.
